gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/filter/AbstractJWTFilter.java

```diff
@@ -300,8 +302,23 @@ protected Subject createSubjectFromToken(final JWT token) throws UnknownTokenExc
 
   public Subject createSubjectFromTokenIdentifier(final String tokenId) throws UnknownTokenException {
     TokenMetadata metadata = tokenStateService.getTokenMetadata(tokenId);
+    String username = null;
     if (metadata != null) {
-      return createSubjectFromTokenData(metadata.getUserName(), null);
+      String type =  metadata.getMetadata(TYPE);
```

Comment:
As you indicated in the PR's description,
> This change introduces only the acceptance of the Knox TokenID and Passcode tokens as CLIENT_ID and CLIENT_SECRET in a standard OAuth 2.0 client credentials flow request body

Based on this, I've to ask if there is going to be a separate PR where the `type` metadata is created when OAuth clients get a new token (and replay the `tokenId` and `passcode` fields as `CLIENT_ID` and `CLIENT_SECRET`)?

Reply:
As we discussed offline, there's going to be a follow-up JIRA to handle the `type` metadata.